sys/net/netif/netif.c

```diff
@@ -65,7 +65,8 @@ netif_t *netif_get_by_name(const char *name)
 
     while (node) {
        netif_get_name((netif_t *)node, tmp);
-       if (strncmp(name, tmp, CONFIG_NETIF_NAMELENMAX) == 0) {
+       size_t len = strlen(tmp);
```

Comment:
`tmp` may not be `\0` terminated, so this does not work.

Reply:
If you want to check for length of `tmp` (which I agree might be necessary, I am afraid you need to do it by hand:

```suggestion
       size_t len;
       
        for (len = 0; len < CONFIG_NETIF_NAMELENMAX; len++) {
            if (tmp[i] == '\0') {
               break;
            }
        }
```

Reply:
Hmm but [the documentation](http://doc.riot-os.org/group__net__netif.html#ga21ea06eb0aa8db28b28a8248bcd2ef18) says:

>Supposed to be implemented by the networking module. name must be zero-terminated in the result!

Reply:
Ok, then I remembered that wrong (and the usage of `CONFIG_NETIF_NAMELENMAX` with the original `strncmp()` is not really clear to me then either).